gpytorch/kernels/multitask_kernel.py

```diff
@@ -38,21 +38,24 @@ def forward_diag(self, x1, x2):
         in :class:`gpytorch.kernels.Kernel` because we need to take the Kronecker product of the diagonals of the
         base data kernel and the task kernel.
         """
-        x1 = x1.transpose(-2, -3)
-        x2 = x2.transpose(-2, -3)
         task_indices = torch.arange(self.n_tasks, device=x1.device).long()
         task_indices = task_indices.unsqueeze(-1).unsqueeze(-1)
 
         # These are small because they are vectors, therefore it is safe to evaluate them
-        covar_i = self.task_covar_module(task_indices).evaluate().squeeze(-1)
-        covar_x = self.data_covar_module.forward(x1, x2)
-        if isinstance(covar_x, LazyVariable):
-            covar_x = covar_x.evaluate()
-        covar_x = covar_x.squeeze(-1)
+        covar_i_diag = self.task_covar_module.forward_diag(task_indices)

```

Comment:
This raises a `TypeError: forward_diag() missing 1 required positional argument: 'x2'`

Reply:
however, if we replace this by `forward_diag(task_indices, task_indices)` we run into size incompatibilities in `observed_pred.confidence_region()` in the example nb.

Reply:
Sorry about that, should be fixed